### PR TITLE
CLOVER-15679 handle case when no devices are present when trying to make a sale

### PR DESCRIPTION
**Pull request by xianm**
JIRA: https://jira.dev.clover.com/browse/CLOVER-15679

@matthewoates

### CODE REVIEW

**Comment by xianm** [comment]
Screenshot
screen shot 2015-09-30 at 5 20 25 pm

File: src/js/CloverPrivate/components/DevicePicker.js [hunk 1]
```diff
@@ -92,7 +97,21 @@ var DevicePicker = React.createClass({
...
+          return this.state.devices.map(this.renderDevice);
+      } else {
+          return (
+              <div>
+                  <p>You do not have any supported devices.</p>
+
+                  <FlatButton
+                      label="close"
+                      secondary={true}
+                      onClick={this.close}
```
Comment (matthewoates) [inline src/js/CloverPrivate/components/DevicePicker.js]: `CloverRunner.close`

It handles a few more things like sending a reject message to the parent window, and gracefully closing the device connection.

:ok_hand:

Comment (xianm) [inline src/js/CloverPrivate/components/DevicePicker.js]: Should `CloverRunner.closeClover()` be added to the `close()` function?

i.e.:

```    close: function () {
      CloverRunner.closeClover()
      close();
    },
```

**Comment by matthewoates** [comment]
:ok_hand:

**Pull request merged**